File: 1955b/main.py
def I():
    return input()


def II():
    return int(input())


def MII():
    return map(int, input().split())


def LI():
    return list(input().split())


def LII():
    return list(map(int, input().split()))
from collections import Counter
# alpha = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
a = II()
def solve():
    n, c, d = MII()

    L = LII()
    x = min(L)
    dic = Counter(L)

    M = []
    for i in range(n):
        M.append(i * c + x)

    for i in M:
        for j in range(n):
            dic[d * j + i] -= 1
            if dic[d * j + i] < 0:
                print("NO")
                return
    else:
        print("YES" if all(dic[i] == 0 for i in dic) else 'NO')
for _ in range(a):
    solve()

File: 1955b/main.go
package main

import (
	"bufio"
	. "fmt"
	"io"
	"os"
)

func solve(in *bufio.Reader, out *bufio.Writer) {
	var n, c, d int
	Fscan(in, &n, &c, &d)

	dic := make(map[int]int)
	M := make([]int, n)
	x := 10_000_000_000
	var tmp int
	for i := 0; i < n*n; i++ {
		Fscan(in, &tmp)
		x = min(x, tmp)
		dic[tmp] += 1
	}

	for i := 0; i < n; i++ {
		M[i] = i*c + x
	}

	for _, a := range M {
		for j := 0; j < n; j++ {
			dic[d*j+a] -= 1
			if dic[d*j+a] < 0 {
				Fprintln(out, "NO")
				return
			}
		}
	}

	for _, v := range dic {
		if v != 0 {
			Fprintln(out, "NO")
			return
		}
	}
	Fprintln(out, "YES")
	return
}

func run(r io.Reader, w io.Writer) {
	in := bufio.NewReader(r)
	out := bufio.NewWriter(w)
	defer out.Flush()
	var a int
	Fscan(in, &a)
	for i := 0; i < a; i++ {
		solve(in, out)
	}

}

func main() {
	run(os.Stdin, os.Stdout)
}
